### PR TITLE
test: identity migrations can be QA tested

### LINKED ISSUE #25948: Integration tests for the identity migration
The Identity migration app is not tested in integration tests yet. As this application becomes more and more complex, adding integration tests early on helps building confidence and makes iteration faster by removing the need for manual testing.

The Identity migration has several moving parts that need to be available:
- Old identity
  - But not too old, we need new endpoints such as https://github.com/camunda-cloud/identity/pull/3223
- New Zeebe with authorization enabled
- ES being filled by the new Camunda Exporter
- New migration app

### PR DESCRIPTION
**Pull request by lenaschoenburg**
This adds a first QA test for the identity migration, asserting that it can migrate the default entities without crashing.

Besides a couple of minor fixes to the identity migration, this changes how migrations are run overall: 3b503cca0e910c56e424a8b978888602a1d51df0. @panagiotisgts Your review on this commit alone would be appreciated!

Closes https://github.com/camunda/camunda/issues/25948

### CODE REVIEW

File: qa/integration-tests/src/test/java/io/camunda/it/migration/IdentityDefaultEntitiesIT.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        .withAdditionalProfile(Profile.IDENTITY_MIGRATION)
+        .withProperty("camunda.migration.identity.zeebe.gatewayAddress", camunda.restAddress())
+        .withProperty(
+            "camunda.migration.identity.managementIdentity.baseUrl", externalIdentityUrl(identity))
+        .withProperty(
+            "camunda.migration.identity.managementIdentity.m2mToken",
+            getIdentityAccessToken(keycloak))
+        .start();
+
+    // then -- the migration did not fail
```
Comment (maryarm) [inline qa/integration-tests/src/test/java/io/camunda/it/migration/IdentityDefaultEntitiesIT.java:54]: I assume later we can add more asserts by checking ES and management identity migration reporting endpoints to check for the migrated records but for now It's ok.

**Review by maryarm — APPROVED** [review]
LGTM 👍

**Review by panagiotisgts — APPROVED** [review]
Changes LGTM, left one question for @maryarm in the code.

Also, something I might have missed in the previous PR of Maryam, the `application-migration.properties` files should be outside the `META-INF` resource directory, to be picked up by the preset config location scans. If you can include this in your changelog @lenaschoenburg .

File: migration/migration-api/src/main/java/io/camunda/migration/api/Migrator.java [hunk 2]
```diff
@@ -7,17 +7,7 @@
...
 
 /** Scanned interface for migration tasks. */
-public interface Migrator extends Runnable {
-
-  default void run(final ApplicationArguments args) {
-    acceptArguments(args);
-    run();
-  }
-
-  default void acceptArguments(final ApplicationArguments args) {
```
[the remark this answers is not in the document]
Comment (lenaschoenburg) [reply migration/migration-api/src/main/java/io/camunda/migration/api/Migrator.java:20]: There was some code suggesting a "status" command but that was not implemented so I removed it. If we actually need it, we can have `MigrationsRunner` pass `ApplicationArguments` again.

**Pull request merged**